File: FurFriendzApp/api/model/petSitterDTO.ts
import AddressModel from './addressModel'
import ServiceDTO from './serviceDTO'

export interface PetSitterDTO {
    id: string; // Guid is equivalent to string in TypeScript
    lastName: string;
    firstName: string;
    username: string;
    telephone?: string; // Optional field
    email?: string; // Optional field
    imageId?: Uint8Array; // byte[] in C# is Uint8Array in TypeScript
    homeAddress?: AddressModel; // Optional field
    services: ServiceDTO[]; // List<ServiceDTO> is ServiceDTO[]
}

File: Service/Models/PetSitterDTO.cs
using Service.Types;

namespace Service.Models;

public class PetSitterDTO
{
	public Guid Id { get; set; }
	public required string LastName { get; set; }
	public required string FirstName { get; set; }
	public required string Username { get; set; }
	public string? Telephone { get; set; }
	public string? Email { get; set; }
	public byte[]? ImageID { get; set; }
	public string? Description { get; set; }
	public bool? isVerified { get; set; }
	public AddressDTO? HomeAddress { get; set; }
	public List<ServiceDTO> Services { get; set; }
}
